src/agentlab/llm/chat_api.py

```diff
@@ -1,28 +1,23 @@
+import logging
 import os
 import re
+import time
 from abc import ABC, abstractmethod
 from dataclasses import dataclass
-from typing import TYPE_CHECKING
 
-from langchain.schema import AIMessage
+import openai
+from openai import AzureOpenAI, OpenAI
 
 import agentlab.llm.tracking as tracking
-from agentlab.llm.langchain_utils import (
-    ChatOpenRouter,
-    HuggingFaceAPIChatModel,
-    HuggingFaceURLChatModel,
-    _convert_messages_to_dict,
-)
-
-if TYPE_CHECKING:
-    from langchain_core.language_models.chat_models import BaseChatModel
+from agentlab.llm.huggingface_utils import HuggingFaceURLChatModel
+from agentlab.llm.llm_utils import _extract_wait_time
 
 
 class CheatMiniWoBLLM:
     """For unit-testing purposes only. It only work with miniwob.click-test task."""
 
     def invoke(self, messages) -> str:
-        prompt = messages[-1].content
+        prompt = messages[-1].get("content", "")
```

Comment:
Let's avoid silent fail here.

```suggestion
        prompt = messages[-1]["content"]
```